include/nanobind/nb_class.h

```diff
@@ -376,6 +376,29 @@ template <typename Arg> struct init_implicit {
     }
 };
 
+template <typename Func, typename Sig = detail::function_signature_t<Func>>
+struct new_;
+
+template <typename Func, typename Return, typename... Args>
+struct new_<Func, Return(Args...)> {
+    std::remove_reference_t<Func> func;
+
+    new_(Func &&f) : func((detail::forward_t<Func>) f) {}
+
+    template <typename Class, typename... Extra>
+    NB_INLINE void execute(Class &cl, const Extra&... extra) {
+        using Type = typename Class::Type;
+        cl.def_static(
+            "__new__",
+            [func = (detail::forward_t<Func>) func](type_object_t<Type>, Args... args) {
+                return func((detail::forward_t<Args>) args...);
+            },
+            extra...);
+        cl.def("__init__", [](handle_t<Type>, Args...) {}, extra...);
```

Comment:
can this just take a `nb::handle` for ``self``? I don't see the harm, and this avoids a type check done by nanobind.

Reply:
Yep, I was trying to get the signature correct but as you point out it doesn't matter.